dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/rest/TelemetryServletRequestListener.java

```diff
@@ -23,15 +26,22 @@
  */
 public class TelemetryServletRequestListener implements ServletRequestListener {
 
+    private static final TraceComponent tc = Tr.register(TelemetryServletRequestListener.class);
+
     /** {@inheritDoc} */
     @Override
     public void requestDestroyed(ServletRequestEvent sre) {
+        try {
         // End the span scope (if present)
-        Scope scope = (Scope) sre.getServletRequest().getAttribute(TelemetryContainerFilter.SPAN_SCOPE);
-        if (scope != null) {
-            scope.close();
+            Scope scope = (Scope) sre.getServletRequest().getAttribute(TelemetryContainerFilter.SPAN_SCOPE);
+            if (scope != null) {
+
+                scope.close();
+            }
             sre.getServletRequest().removeAttribute(TelemetryContainerFilter.SPAN_SCOPE);
+        }  catch (Exception e) {
```

Comment:
Previously `sre.getServletRequest().removeAttribute(TelemetryContainerFilter.SPAN_SCOPE);` was inside the `if` block, now it is outside.